src/main/java/org/mitre/synthea/engine/State.java

```diff
@@ -590,14 +588,6 @@ protected void initialize(Module module, String name, JsonObject definition) {
     @Override
     public SetAttribute clone() {
       SetAttribute clone = (SetAttribute) super.clone();
```

Comment:
Based on the comment in the original should we be setting threadExpProcessor to null?

Reply:
Good catch - I meant to explicitly call that out but forgot. Actually the original comment was wrong. We definitely don't want to copy thread-specific variables, but the whole point of the `ThreadLocal` class is that it *is* safe to pass around between threads, because it's a wrapper around the object, and it uses an internal map of Thread->object to ensure that each thread only sees its own version of the object.

Leaving it as-is means that the underlying Object.clone is copying the field and the object is passed around, just as it was before. It won't break anything if we do set it to null, but I think the intent of using the ThreadLocal in the first place was that we want to reuse the underlying object because it's expensive to create, but the libraries it uses aren't threadsafe, so we only want to use it within the same thread. So if we set `threadExpProcessor = null` in clone(), I don't think there's any reason to use ThreadLocal at all. I looked at this in detail a few months ago, trying to make CQL processing less resource-intensive, but I don't remember the details. I think it's worth revisiting, though my vote would be to leave this as is today